### PR TITLE
Support JSON API 1.0.

### PR DESCRIPTION
**Pull request by barelyknown**
I removed the extract links tests (they're no longer relevant) and made the rest of the tests pass. I didn't refactor much, but it looks like there is a decent amount of code that is no longer necessary. I also think that the normalize and serialize flow is strange as is (I probably should have rewritten the normalize method but didn't have time to dig in to understand enough about how the library works).

I'd be happy to clean it up more if you'd like (once I have time), or feel free to take it and run with it.

### CODE REVIEW

**Comment by andrewbranch** [comment]
I forked this and confirmed it works when changed to `data: data`.

**Comment by barelyknown** [comment]
Yeah - the `data: data` change was my mistake. Merged now.

**Comment by mrleeio** [comment]
Random newbie Ember Data question: I read on the ember data issue tracker that it plans on supporting json api out of the box.  Does this mean that this adapter will not be required or is it somehow specific for jsonapi-resources?

**Comment by barelyknown** [comment]
I asked about out-of-the-box support for JSON API on the Ember Slack channel yesterday, and they confirmed that that was the plan. But, I don't know any more specifics. I assume/hope that this library won't be necessary once it's the default.

File: tests/integration/specs/href-link-for-resource-collection-test.js [hunk 1]
```diff
@@ -1,137 +1,154 @@
...
-        equal(someResources.get('length'), 2, 'there are 2 someResources');
-
-        equal(something1.get('title'), 'Something 1', 'something1 title');
-        equal(something2.get('title'), 'Something 2', 'something2 title');
-        start();
-      });
-    });
-  });
-});
+// var get = Ember.get, set = Ember.set;
```
Comment (eneuhauser) [inline tests/integration/specs/href-link-for-resource-collection-test.js:1]: Why is this whole file commented out?

File: tests/integration/specs/multiple-resource-links-test.js [hunk 2]
```diff
@@ -11,43 +11,57 @@ module('integration/specs/multiple-resource-links-test', {
...
-                                id: '2'
-                            }
+                    attributes: {
+                      title: 'Rails is Omakase',
+                    },
+                    relationships: {
+                      author: {
+                        links: {
+                          self: '/posts/1/relationships/author',
+                          related: '/posts/1/author',
```
Comment (eneuhauser) [inline tests/integration/specs/multiple-resource-links-test.js:21]: Trailing comma

File: tests/integration/specs/multiple-resource-links-test.js [hunk 3]
```diff
@@ -65,31 +79,33 @@ module('integration/specs/multiple-resource-links-test', {
...
+           var post2 = records.objectAt(1);
+           var promises = [];
 
-             promises.push(post1.get('author').then(function(author) {
-                equal(author.get('name'), 'dhh', 'post1 author');
-             }));
-             promises.push(post2.get('author').then(function(author) {
-                 equal(author.get('name'), 'ado', 'post2 author');
-             }));
+           promises.push(new Promise(function(resolve, reject) { resolve('asdf'); }));
```
Comment (eneuhauser) [inline tests/integration/specs/multiple-resource-links-test.js:99]: This fails when running `npm run test` because `Promise` is not found. Needs to be `Ember.RSVP.Promise` instead.

File: tests/unit/serializer/extract-links-test.js [hunk 4]
```diff
@@ -1,121 +0,0 @@
-var serializer;
```
Comment (eneuhauser) [inline tests/unit/serializer/extract-links-test.js:1]: Why was this file deleted?

**Comment by eneuhauser** [comment]
@barelyknown please checkout my comments. Be sure to test with `npm run test` because that's what the travis build runs to ensure the merge.

I'm currently upgrading my server to be 1.0 compliant. I'll verify this build with my server for additional tests.

File: tests/integration/specs/href-link-for-resource-collection-test.js (hunk 1 above)
Comment (barelyknown) [inline tests/integration/specs/href-link-for-resource-collection-test.js:1]: Shouldn't have been I don't think. I probably just forgot to add it back as I worked through the issues.

**Comment by barelyknown** [comment]
I haven't looked at this in about a week, but I should later this week. Since ember-data is planning on shipping a JSON API 1.0 compliant serializer and adapter shortly (I just noticed that it was merged into master ~2 days ago), do you plan on maintaining this?

**Comment by miguelsan** [comment]
It'd be good to ship this asap. Otherwise we'll have to cope with the unlucky current situation where the spec is in v1.0, ember data is far from shipping (see emberjs/data/issues/2905), and ember-json-api supports only an outdated, non standard version. This can break apps anywhere :-1:

**Comment by barelyknown** [comment]
`ember-data` is pretty close to being compliant: https://github.com/emberjs/data/pull/2904

**Comment by kurko** [comment]
This could be opened against the main repo, specially now that @eneuhauser is a core member ☺️

**Comment by jonkoops** [comment]
I hate to be this guy but **+1**. We should get this merged against the main repo.
